app/src/main/java/com/swent/assos/model/view/EventViewModel.kt

```diff
@@ -186,4 +190,22 @@ constructor(
   fun setStaffingEnabled(isEnabled: Boolean) {
     _event.value = _event.value.copy(isStaffingEnabled = isEnabled)
   }
+
+  fun saveEvent(eventId: String) {
+    DataCache.currentUser.value =
+        DataCache.currentUser.value.copy(
+            savedEvents = DataCache.currentUser.value.savedEvents + eventId)
+    viewModelScope.launch(ioDispatcher) {
+      dbService.saveEvent(eventId, { _isSaved.update { true } }, {})
+    }
+  }
+
+  fun unSaveEvent(eventId: String) {
+    DataCache.currentUser.value =
+        DataCache.currentUser.value.copy(
+            savedEvents = DataCache.currentUser.value.savedEvents.filter { it != eventId })
```

Comment:
It could maybe be a good idea to fetch only the events that has this particular eventId so that we do not fetch every events everytimie.